File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.cc
#include<iostream>
#include<math.h>
using namespace std;

char arr[55][55];

int n, m;
int val;
int ans = 32;
int toggle = 1; //w

bool inrange(int x, int y) {
	return x >= 0 && x < n && y >= 0 && y < m;
}

void check(int x, int y) {
	for (int i = x; i < x + 8; i++) {
		for (int j = y; j < y + 8; j++) {
			if (toggle == 1) {
				if (arr[i][j] != 'W') val++;
			}
			else {
				if (arr[i][j] != 'B') val++;
			}
			toggle = -toggle;
		}

		toggle = -toggle;
	}
}


int main()
{
	ios::sync_with_stdio(0); cin.tie(0);
	cin >> n >> m;

	for (int i = 0; i < n; i++) {
		string s; cin >> s;
		for (int j = 0; j < m; j++) arr[i][j] = s[j];
	}

	for (int i = 0; i < n; i++) {
		for (int j = 0; j < m; j++) {
			if (inrange(i + 7, j + 7)) {
				toggle = 1; val = 0;
				check(i, j);
				ans = min(ans, val);

				toggle = -1; val = 0;
				check(i, j);
				ans = min(ans, val);
			}
		}
	}

	cout << ans;


	return 0;
}

File: 백준/Silver/1018. 체스판 다시 칠하기/체스판 다시 칠하기.java
import java.io.*;
import java.util.*;

class Main {
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(br.readLine());
		int n = Integer.parseInt(st.nextToken());
		int m = Integer.parseInt(st.nextToken());
		
		char[][] charArr = new char[n][m];
		
		for(int i = 0; i < n; i++){
			String s = br.readLine();
			char[] tmpArr = s.toCharArray();
			for(int j = 0; j < m; j++)
				charArr[i][j] = tmpArr[j];
		}
		
		int ans = 2500;
		
		for(int i = 0; i < n - 7; i++){
			for(int j = 0; j < m - 7; j++){
				int tmpMin = Math.min(checkWBVer1(charArr, i, j),checkWBVer2(charArr, i, j));
				ans = Math.min(ans,tmpMin);
			}
		}
		
		System.out.println(ans);
	}
	
	public static int checkWBVer1(char[][] arr, int x, int y){
		int cnt = 0;
		for(int i = x; i < x + 8; i++){
			for(int j = y; j < y + 8; j++){
				if(i % 2 == 0 && j % 2 == 1){
					if(arr[i][j] != 'W') cnt++;
				}else if(i % 2 == 0 && j % 2 == 0){
					if(arr[i][j] != 'B') cnt++;
				}else if(i % 2 == 1 && j % 2 == 0){
					if(arr[i][j] != 'W') cnt++;
				}else{
					if(arr[i][j] != 'B') cnt++;
				}
			}
		}
		return cnt;
	}
	
	public static int checkWBVer2(char[][] arr, int x, int y){
		int cnt = 0;
		for(int i = x; i < x + 8; i++){
			for(int j = y; j < y + 8; j++){
				if(i % 2 == 0 && j % 2 == 1){
					if(arr[i][j] != 'B') cnt++;
				}else if(i % 2 == 0 && j % 2 == 0){
					if(arr[i][j] != 'W') cnt++;
				}else if(i % 2 == 1 && j % 2 == 0){
					if(arr[i][j] != 'B') cnt++;
				}else{
					if(arr[i][j] != 'W') cnt++;
				}
			}
		}
		return cnt;
	}
}